modules/acm/src/test/scala/edu/gemini/epics/acm/ObserveStateSpec.scala

```diff
@@ -0,0 +1,402 @@
+    // CAR VAL change
+    observe.onCarValChange(CarState.IDLE)
+    assert(!observe.applyState().isIdle())
+
+    // OBSERVE
+    // OBSERVE goes BUSY
+    // VAL change
+    observe.onApplyValChange(359)
+    assert(!observe.applyState().isIdle())
+    // Observe CAR VAL change
+    observe.onObserveCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(359)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+
+    // OBSERVE goes IDLE
+    // Observe CAR VAL change
+    observe.onObserveCarValChange(CarState.IDLE)
+    // And we are done and IDLE
+    assert(observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(359)
+    assert(observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.IDLE)
+    assert(observe.applyState().isIdle())
+  }
+
+  test("GMOS normal observation") {
+    val context: CAJContext = mock[CAJContext]
+    (context.addContextExceptionListener _).expects(*).returns(()).repeat(5)
+    (context.addContextMessageListener _).expects(*).returns(()).repeat(5)
+    (context.pendIO _).expects(*).returns(()).repeat(1 to 6)
+    // We just return null as we don't need the channels and don't want to mock them
+    (context.createChannel(_: String)).expects("gm:apply.DIR").returns(null)
+    (context.createChannel(_: String)).expects("gm:apply.VAL").returns(null)
+    (context.createChannel(_: String)).expects("gm:apply.MESS").returns(null)
+    (context.createChannel(_: String)).expects("gm:applyC.CLID").returns(null)
+    (context.createChannel(_: String)).expects("gm:applyC.VAL").returns(null)
+    (context.createChannel(_: String)).expects("gm:applyC.OMSS").returns(null)
+    val epicsService = new EpicsService(context)
+    val observe = new CaObserveSenderImpl(
+      "gmos::observeCmd",
+      "gm:apply",
+      "gm:applyC",
+      "gm:dc:observeC",
+      "gm:stop",
+      "gm:abort",
+      "GMOS Observe",
+      classOf[CarState],
+      epicsService)
+    // Start idle
+    assert(observe.applyState().isIdle())
+    // Post an observe
+    observe.post()
+
+    // CONFIGURATION
+    // APPLY Goes to busy
+    // VAL change
+    observe.onApplyValChange(4166)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4166)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.BUSY)
+    assert(!observe.applyState().isIdle())
+
+    // APPLY Goes to IDLE
+    // Another VAL change
+    observe.onApplyValChange(4166)
+    assert(!observe.applyState().isIdle())
+    // CAR CLID change
+    observe.onCarClidChange(4166)
+    assert(!observe.applyState().isIdle())
+    // CAR VAL change
+    observe.onCarValChange(CarState.IDLE)
+    assert(!observe.applyState().isIdle())
+
+    // OBSERVE
```

Comment:
The observe must be triggered by a call to `post` in the `ObserveSenderImpl` instance.